cmake/sysbuild/partition_manager.cmake

```
@@ -0,0 +1,835 @@
+  set_shared(IMAGE ${DOMAIN} PROPERTY PM_HEX_FILE ${PROJECT_BINARY_DIR}/${merged}.hex)
+  set_shared(IMAGE ${DOMAIN} PROPERTY PM_DOTCONF_FILES ${pm_out_dotconf_file})
+  set_shared(IMAGE ${DOMAIN} PROPERTY PM_APP_HEX ${PROJECT_BINARY_DIR}/app.hex)
+  set_shared(IMAGE ${IMAGE_NAME} APPEND PROPERTY BUILD_BYPRODUCTS ${PROJECT_BINARY_DIR}/${merged}.hex)
+  if(CONFIG_SECURE_BOOT)
+    # Only when secure boot is enabled the app will be signed.
+    set_shared(IMAGE ${DOMAIN} PROPERTY PM_SIGNED_APP_HEX ${PROJECT_BINARY_DIR}/signed_by_b0_app.hex)
+  endif()
+else()
+  # This is the root image, generate the global pm_config.h
+  # First, include the shared properties for all child images.
+  if (PM_DOMAINS)
+    # We ensure the existence of PM_DOMAINS to support older cmake versions.
+    # When version >= 3.17 is required this check can be removed.
+    list(REMOVE_DUPLICATES PM_DOMAINS)
+  endif()
+  foreach (d ${PM_DOMAINS})
+    # Don't include shared vars from own domain.
+    if (NOT ("${DOMAIN}" STREQUAL "${d}"))
+      get_shared(shared_header_files          IMAGE ${d} PROPERTY PM_DOMAIN_HEADER_FILES)
+      get_shared(shared_prefixed_images       IMAGE ${d} PROPERTY PM_DOMAIN_IMAGES)
+      get_shared(shared_pm_out_partition_file IMAGE ${d} PROPERTY PM_DOMAIN_PARTITIONS)
+      get_shared(shared_pm_out_region_file    IMAGE ${d} PROPERTY PM_DOMAIN_REGIONS)
+      get_shared(shared_domain_hex_files      IMAGE ${d} PROPERTY PM_HEX_FILE)
+
+      list(APPEND header_files          ${shared_header_files})
+      list(APPEND prefixed_images       ${shared_prefixed_images})
+      list(APPEND pm_out_partition_file ${shared_pm_out_partition_file})
+      list(APPEND pm_out_region_file    ${shared_pm_out_region_file})
+      list(APPEND domain_hex_files      ${shared_domain_hex_files})
+      list(APPEND global_hex_depends    ${${d}_PM_DOMAIN_DYNAMIC_PARTITION}_subimage)
+
+      # Updated code.
+      # All partition files are now created in this CMake invocation, so just add them:
+      list(APPEND pm_out_partition_file ${APPLICATION_BINARY_DIR}/partitions_${d}.yml)
+      list(APPEND pm_out_region_file    ${APPLICATION_BINARY_DIR}/regions_${d}.yml)
+
+      # Add domain prefix cmake variables for all partitions
+      # Here, we actually overwrite the already imported kconfig values
+      # for our own domain. This is not an issue since all of these variables
+      # are accessed through the 'partition_manager' target, and most likely
+      # through generator expression, as this file is one of the last
+      # cmake files executed in the configure stage.
+      import_kconfig(PM_ ${APPLICATION_BINARY_DIR}/pm_${d}.config ${d}_pm_var_names)
+
+      foreach(name ${${d}_pm_var_names})
+        set_property(
+          TARGET partition_manager
+          PROPERTY ${d}_${name}
+          ${${name}}
+          )
+      endforeach()
+    endif()
+  endforeach()
+
+  if (CONFIG_BOOTLOADER_MCUBOOT)
```

Kind of a corner case, but I get an error when building with `-DCONFIG_BOOTLOADER_MCUBOOT=y` (with or without the `SB_` version)
```
CMake Error at /home/grsw/ncs/nrf/cmake/sysbuild/partition_manager.cmake:748 (math):
  math cannot parse the expression: " + - ": syntax error, unexpected end of
  file (18).
```
Remember that `CONFIG_BOOTLOADER_MCUBOOT` is read from sysbuild's own cache, which is not in sync with the main app config.

I wonder whether this should be:
```suggestion
  if (SB_CONFIG_BOOTLOADER_MCUBOOT)
```

But I see that this code path is not fully ported to sysbuild anyway. There might be some other quirks related to misplaced `CONFIG_*` variables.

---

Correct.
This code is kept as reference / to make it easier to know when and where the parent / child images were moving images.

It is related multi image updates and thus not a goal in this task.

I will comment out the code, but keep it as a reference to facilitate followup task related to image update.